lib/builder/util.go

```diff
@@ -63,15 +65,25 @@ func checkBuildEnv() error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	_, err = client.Version()
-	if err != nil {
-		log.Error(trace.DebugReport(err))
-		return trace.BadParameter(
-			`Docker does not seem to be available on this machine.
+	dockerErr := trace.BadParameter(
+		`Docker does not seem to be available on this machine.
 To resolve the issue:
   * Install Docker (https://docs.docker.com/engine/installation/).
   * Make sure it can be used by a non-root user (https://docs.docker.com/install/linux/linux-postinstall/).`)
+
+	_, err = client.Version()
+	if err != nil {
+		log.Error(trace.DebugReport(err))
+		return dockerErr
+	}
+	_, err = client.Version()
+
+	err = utils.Exec(exec.Command("docker", "help"), ioutil.Discard)
```

Comment:
What's the purpose of executing this command?

Reply:
The purpose of executing this command is to pre-emptively find issues with the docker binary before hitting the code paths in `lib/docker/docker.go` mid tele build.  Its the same reason we call `client.Version()` earlier in `CheckBuildEnv`, but that check fails to validate the binary.  We want to validate both paths, as tele build both uses the native golang docker client and execs the docker binary.  I wanted something quick, since this is pre-build validation logic -- alternatively, we could do `docker version` here.

Reply:
I improved the error message for this in 2ae2d38, which helps explain why this command is being run.